### DISCUSSION TITLE
2009/09/22/%E3%80%8A%E6%88%91%E5%92%8Clabview%E3%80%8B%E4%B9%A6%E8%AF%84%E5%8F%8A%E5%8F%8D%E9%A6%88%E6%84%8F%E8%A7%81

### DISCUSSION THREAD

**Discussion by giscus** [ruanqizhen/blog#280] [Announcements]
# 2009/09/22/%E3%80%8A%E6%88%91%E5%92%8Clabview%E3%80%8B%E4%B9%A6%E8%AF%84%E5%8F%8A%E5%8F%8D%E9%A6%88%E6%84%8F%E8%A7%81

如果大家对本书有任何意见和建议可以直接在这里或下列链接中的任意一个回帖，我会一一解答。由于不是每天都有时间更新博客，所以提出的问题可能会延迟一两天才回答，希望大家谅解。下面的列表是我收集的其它网站上有关本书的评论：

https://blog.qizhen.xyz/2009/09/22/%E3%80%8A%E6%88%91%E5%92%8Clabview%E3%80%8B%E4%B9%A6%E8%AF%84%E5%8F%8A%E5%8F%8D%E9%A6%88%E6%84%8F%E8%A7%81

**Comment by ruanqizhen** [comment]
之前博客下的留言：

Qizhen
2009年12月20日 21:46 编辑
vi.lib 文件下的大多数VI还是开源的。不过我也不没时间一个个都把他们研究一遍。

回复
帆
2009年12月19日 23:49 编辑
P58页提到的 LabVIEW的目录下\\vi.lib\\Utility\\VariantDataType\\GetTypeInfo.vi 这个VI的功能是输入变体类型数据，可以得到变体是何种数据转化的。说这类VI的大部分公开源代码的，但是我在哪个目录里面没有找的几个代码公开的，全是需要密码的。能不能找几个公开代码的例子？

回复
典钢
2009年10月28日 17:16 编辑
阮先生您好，我有一个问题想请教您：我想修改XY图形控件曲线宽度，但是我一共有两条曲线，利用属性结点程序只能修改第一条曲线。请问是什么原因呢？相关附件我贴在网站上，您有空的时候顺带看一下，谢谢！http://bbs.gsdzone.net/showtopic-1180.aspx

回复
典钢
2009年10月24日 17:20 编辑
阮先生您好，我把问题贴上去了，地址：http://bbs.gsdzone.net/showtopic-1169.aspx对不起，这几天请假回家了，没有及时回复。引用的作用是否还可以避免数据竞争？

回复
Qizhen
2009年10月19日 09:09 编辑
你把程序张贴在这个论坛上吧：http://bbs.gsdzone.net/showforum-2.aspx这是我和其他一些NI同事上的比较多的论坛。一旦数据线出现分叉，数据就编程完全独立的两份了。所以，实际上程序是在两个不同线程中对两个完全没有关系的数据进行修改。在程序的最终，你可以按照自己定义的逻辑，再把这两份完全不同的数据合并起来。如果你希望程序在不同线程内对同一数据进行读写，则必须使用传引用的方式，把数据的引用传递到不同的线程里去。

回复
典钢
2009年10月17日 08:26 编辑
阮先生您好，您说的用全局变量终止循环，可是我试过了，不能终止。不知是否哪里写错了。谢谢我怎么把我的文件传给您呢？另外有一个问题想请教：在第11章中代码风格与优化，您讲到为了防止数据被修改，Labview会在节点处进行备份，使不同线程之间的数据不会相互影响！我的问题是如果两个线程都对数据进行了修改，而且是修改的不同的地方，那么如何得到最终的结果呢？

回复
王
2009年10月12日 23:23 编辑
第一时间从卓越网上买到此书（没阮先生签名，表示下遗憾）；一开始只是草草地浏览一遍，突然意识到自己太浮躁了；然后逼迫自己从第一页开始一字一句的认真阅读（因为相信阮先生的功力），现在看完前三章，总体感觉不错，作者写出了很多新手老手都容易忽略的编程“死角”！ 目前国内非常缺乏这种LabVIEW经验心得型的书籍（在这方面，国外也没多少高质量的，The LabVIEW Style Book除外），大部分还处于类LabVIEW Help的“教程”阶段。阮先生算是在国内开先河，希望再接再厉，能有后续作品出版，能为国内打造出一批LabVIEW高手。 同时，也希望高先生和陈老师的作品能早日上架。

回复
Qizhen
2009年10月12日 22:49 编辑
出版社有一些带签名的书，是专门卖给合作网站的网友的。你可以到这些合作网站购买，不过要等到他们搞活动的时候才能买到。你可以关注一下，下面这三处链接：http://ruanqizhen.eefocus.com/http://group.ednchina.com/2335/http://digital.ni.com/worldwide/china.nsf/web/all/B8945411CFD4AC74482574C200241381?OpenDocument&node=207148_zhs

回复
fei
2009年10月12日 14:51 编辑
不好意思，弱弱的问一句。哪里可以买到签名书啊？很想买一本啊。

回复
Qizhen
2009年10月12日 12:54 编辑
如果程序中还有其他并行运行的循环。用户选择关闭后，必须把所有循环的结束条件设为“真”。 最简单的例子，一个循环可以由一个全局变量控制，当用户选择关闭前面板，就改变全局变量的值，另循环停止。

回复
典钢
2009年10月12日 11:45 编辑
阮先生您好，您书中第十章讲到应用程序的退出按钮，可以使用“前面板关闭？”过滤型事件处理。我运行了一下，如果添加了这个事件，在对话框中“是否真的要退出程序”选择是，程序把前面板关闭之后直接结束，后序的代码并不会执行。所以如果我的主vi中有循环，生成exe文件之后，我的循环还是跳不出去。不知您是如何解决的呢？热切盼望着您的答复！

回复
帆
2009年09月25日 16:59 编辑
高先生还是写书吧！居然见到牛人的对话了

回复
Qizhen
2009年09月25日 13:02 编辑
高先生，您过奖了。我也只研究过LabVIEW中的一小部分，写到某些章节时，我也不是太了解的，比如与硬件的关联等，写的都比较简略。相信您和其它LabVIEW用户在各自专研的领域都比我更精通，所以我也很期待您的大作问世。

回复
巍
2009年09月25日 10:39 编辑
阮先生，感谢你和你的同事们！你签名的书已经收到了，正在认真拜读！看了你书中的内容，我都不敢再谈写书一事！labview7i

回复
